scholia/query.py

```diff
@@ -809,6 +809,107 @@ def q_to_label(q, language='en'):
         return None
 
 
+def q_to_label_and_description(q, language='en'):
+    """Get label and description for Q item.
+
+    Parameters
+    ----------
+    q : str
+        String with Wikidata Q item.
+    language : str
+        String with language identifier
+
+    Returns
+    -------
+    label : str
+        String with label corresponding to Wikidata item.
+    description : str
+        String with the description of the corresponding Wikidata item.
+    """
+    query = """SELECT ?label ?description WHERE {{
+          wd:{q} rdfs:label ?label .
+          wd:{q} schema:description ?description .
+          FILTER (LANG(?label) = "{language}")
+          FILTER (LANG(?description) = "{language}")
+        }}""".format(q=q, language=language)
+
+    url = 'https://query.wikidata.org/sparql'
+    params = {'query': query, 'format': 'json'}
+    response = requests.get(url, params=params, headers=HEADERS)
+    data = response.json()
+
+    results = data['results']['bindings']
+    if not results:
+        return {}
+    else:
+        return {
+            "label": results[0]['label']['value'],
+            "description": results[0]['description']['value']
+        }
+
+
+def properties_for_q(q, props, prefixes=None):
+    """Get the value for a property for Q item.
+
+    Parameters
+    ----------
+    q : str
+        String with Wikidata Q item.
+    props : dictionary
+        Dictionary object with the Wikidata property identifier as key and the
+        matching BioSchemas key as value
+    prefixes : dictionary
+        Dictionary object with the prefixes for the Bioschemas values, where
+        the keys are Wikidata properties and the prefix as value
+
+    Returns
+    -------
+    bioschemas : dictionary
+        Map with Bioschemas key-value pairs, matching the input.
+
+    Examples
+    --------
+    >>> property_for_q('Q47512', {"P235": "inChIKey"})
+    {"inChIKey": "QTBSBXVTEAMEQO-UHFFFAOYSA-N"}
+
+    """
+    optionals = str('\n'.join(list(
+        map(lambda s: f"  OPTIONAL {{ ?item wdt:{s} ?{s} }}", props)
+    )))
```

Comment:
This suggestion does 2 things:
1. It switches to using a list comprehension, which is _much_ more readable than a map/lambda
2. It fixes a logical bug. If you iterate over a dictionary `props`, it will iterate only over the keys. But it looks like you actually want to iterate over the items so you can use the key as the wdt:... and the value as the name in the query itself.
```suggestion
    optionals = '\n'.join(
        f"  OPTIONAL {{ ?item wdt:{key} ?{name} }}"
        for key, name in props.items()
    )
```

Reply:
1. PR welcome.
2. no, it should actually iterate only over the keys, not the values